tests/test_generator.py

```diff
@@ -255,55 +243,110 @@ def test_mkdir_file(generator_instance: Generator, test_dir: str):
+def test_copy_files_valid_dest(
+    dest, source, generator_instance: Generator, test_dir: str
+):
+    to_copy = [CopyOperation(src=file, dest=dest) for file in source]
+    gen = GenerationContext(pathlib.Path(test_dir))
+    cmd_list = generator_instance._copy_files(files=to_copy, context=gen)
     assert isinstance(cmd_list, CommandList)
-    for file, cmd in zip(symlink_files, cmd_list):
-        assert file in cmd.command
-
-
-def test_symlink_directory(generator_instance: Generator, get_gen_symlink_dir):
-    """Test Generator._symlink_files helper function with directory"""
-    files = EntityFiles(symlink=get_gen_symlink_dir)
-    cmd_list = generator_instance._symlink_files(files, generator_instance.root)
-    symlinked_folder = generator_instance.root / os.path.basename(get_gen_symlink_dir)
+    # Extract file paths from commands
+    cmd_src_paths = set()
+    for cmd in cmd_list.commands:
+        # Assert destination path exists in cmd
+        assert create_final_dest(pathlib.Path(test_dir), dest) in cmd.command
+        src_index = cmd.command.index("copy") + 1
+        cmd_src_paths.add(cmd.command[src_index])
+    # Assert all file paths are in the command list
+    file_paths = {str(file) for file in source}
+    assert file_paths.issubset(
+        cmd_src_paths
+    ), "Not all file paths are in the command list"
+
+
+@pytest.mark.parametrize(
+    "dest",
+    (
+        pytest.param(None, id="dest as None"),
+        pytest.param(
+            pathlib.Path("/absolute/path"),
+            id="dest as valid path",
+        ),
+    ),
+)
+@pytest.mark.parametrize("source", ["files", "directory"], indirect=True)
+def test_symlink_files_valid_dest(
+    dest, source, generator_instance: Generator, test_dir: str
+):
+    to_copy = [CopyOperation(src=file, dest=dest) for file in source]
+    gen = GenerationContext(pathlib.Path(test_dir))
+    cmd_list = generator_instance._copy_files(files=to_copy, context=gen)
     assert isinstance(cmd_list, CommandList)
-    assert str(symlinked_folder) in cmd_list.commands[0].command
-
-
-def test_write_tagged_file(fileutils, generator_instance: Generator):
-    """Test Generator._write_tagged_files helper function with file list"""
-    conf_path = fileutils.get_test_conf_path(
-        osp.join("generator_files", "easy", "marked/")
-    )
-    tagged_files = sorted(glob(conf_path + "/*"))
-    files = EntityFiles(tagged=tagged_files)
-    param_set = {
+    # Extract file paths from commands
+    cmd_src_paths = set()
+    for cmd in cmd_list.commands:
+        # Assert destination path exists in cmd
+        assert create_final_dest(pathlib.Path(test_dir), dest) in cmd.command
+        src_index = cmd.command.index("copy") + 1
+        cmd_src_paths.add(cmd.command[src_index])
+    # Assert all file paths are in the command list
+    file_paths = {str(file) for file in source}
+    assert file_paths.issubset(
+        cmd_src_paths
+    ), "Not all file paths are in the command list"
```

Comment:
Similar question here
```suggestion
    assert file_paths == cmd_src_paths, "Not all file paths are in the command list"
```